core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -121,8 +125,7 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
           context.reply(RegisterExecutorFailed("Duplicate executor ID: " + executorId))
         } else {
           logInfo("Registered executor: " + executorRef + " with ID " + executorId)
-          context.reply(RegisteredExecutor)
-
+          context.reply(RegisteredExecutor(latestTokens))
```

Comment:
Correct. If an executor dies after the initial login, a new one starting up would not have tokens to read the data from HDFS. So on registering, they'd get new tokens that can be used to access HDFS.